subworkflows/local/qiime2_preptax.nf

```diff
@@ -2,20 +2,58 @@
  * Training of a classifier with QIIME2
  */
 
+include { UNTAR                 } from '../../modules/nf-core/untar/main'
 include { FORMAT_TAXONOMY_QIIME } from '../../modules/local/format_taxonomy_qiime'
 include { QIIME2_EXTRACT        } from '../../modules/local/qiime2_extract'
 include { QIIME2_TRAIN          } from '../../modules/local/qiime2_train'
 
 workflow QIIME2_PREPTAX {
     take:
     ch_qiime_ref_taxonomy //channel, list of files
+    val_qiime_ref_taxonomy //val
     FW_primer //val
     RV_primer //val
 
     main:
-    FORMAT_TAXONOMY_QIIME ( ch_qiime_ref_taxonomy )
+    if (params.qiime_ref_tax_custom) {
```

Comment:
Why did you choose a tarball instead of e.g. of a comma separated list of compressed files, such as `--qiime2_ref_tax_custom "seq.fna.gz,taxonomy.tax.gz"`? I think the tarball has the disadvantage of requiring un-compressing before checking that files with required regex are present while the list of files can be checked with nextflow code immediately. Is there somewhere a resource for tar'ed QIIME2 databases as for Kraken2?

Reply:
I've added support for a pair of comma separated files. I saw that the unite databases were pulled as a single tarball, and figure users may prepare databases that they want to move around easier as a single file.

I've not added gzip support for files when pointing to a directory to give some better consistency, but I wanted to get some feedback first on if the flexibility of supporting different input forms is desired.

Reply:
Great, I think thats all one can ask for.
It would be optimal to cover as many inputs as possible in tests configs.